src/JsonSchema/Constraints/BaseConstraint.php

```diff
@@ -36,23 +37,28 @@ public function __construct(Factory $factory = null)
         $this->factory = $factory ?: new Factory();
     }
 
-    public function addError(JsonPointer $path = null, $message, $constraint = '', array $more = null)
+    public function addError(JsonPointer $path = null, ConstraintError $constraint = null, array $more = array())
     {
         $error = array(
```

Comment:
`ConstraintError` is just an enum. I'm not sure it's really meant to be built up into a proper class, with it's own members and whatnot. You don't `new` them, for example. You can read up on it here: https://github.com/marc-mabe/php-enum